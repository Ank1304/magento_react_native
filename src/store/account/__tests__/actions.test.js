import {
  logout,
  getOrderList,
  getCurrentCustomer,
  getOrderedProductInfo,
} from '../actions';

describe('account actions', () => {
  test('getCurrentCustomer()', () => {
    // Exercise
    const result = getCurrentCustomer();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('logout()', () => {
    // Exercise
    const result = logout();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getOrderList()', () => {
    // Setup
    const customerId = 2;

    // Exercise
    const result = getOrderList(customerId);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getOrderedProductInfo()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = getOrderedProductInfo(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });
});
