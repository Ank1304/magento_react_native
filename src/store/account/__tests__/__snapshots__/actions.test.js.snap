// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account actions getCurrentCustomer() 1`] = `
Object {
  "type": "MAGENTO_CURRENT_USER_REQUEST",
}
`;

exports[`account actions getOrderList() 1`] = `
Object {
  "payload": Object {
    "customerId": 2,
  },
  "type": "MAGENTO_GET_ORDERS_REQUEST",
}
`;

exports[`account actions getOrderedProductInfo() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "MAGENTO_GET_ORDERED_PRODUCT_INFO_REQUEST",
}
`;

exports[`account actions logout() 1`] = `
Object {
  "type": "USER_LOGOUT",
}
`;
